cluster.go

```diff
@@ -222,10 +217,42 @@ func (c *clusterClient) _refresh() (err error) {
 
 	groups := result.parse(c.opt.TLSConfig != nil)
 	conns := make(map[string]connrole, len(groups))
+	var wg sync.WaitGroup
 	for master, g := range groups {
-		conns[master] = connrole{conn: c.connFn(master, c.opt), replica: false}
+		conns[master] = connrole{
+			conn:    c.connFn(master, c.opt),
+			replica: false,
+		}
 		for _, addr := range g.nodes[1:] {
-			conns[addr] = connrole{conn: c.connFn(addr, c.opt), replica: true}
+			cc := c.connFn(addr, c.opt)
+
+			if !c.opt.ReplicaOnly {
+				conns[addr] = connrole{
+					conn:    cc,
+					replica: true,
+				}
+				continue
+			}
+
+			wg.Add(1)
+			go func(cc conn) {
+				defer wg.Done()
+
+				timeout := c.opt.Dialer.Timeout
+				if timeout <= 0 {
+					timeout = DefaultDialTimeout
+				}
+
+				ctx, cancel := context.WithTimeout(context.Background(), timeout)
+				defer cancel()
+
+				cc.Do(ctx, cmds.NewCompleted([]string{"READONLY"})) // ignore error
+			}(cc)
```

Comment:
Hi @proost,

Thank you for the changes, but this goroutine looks wired to me and I don't expect there will be breaking change by reusing the `ReplicaOnly`. I am afraid we have some misunderstanding here.

What I previously suggested is that the `pipe` will send `READONLY` command already if its `ReplicaOnly` is true.
So I think we can reuse this behavior by modifying the `c.connFn(addr, c.opt)` in the refresh function, and then the above goroutine is not necessary.

Reply:
Given that the signature of the `connFn` is `func(dst string, opt *ClientOption) conn` which accepts a ClientOption pointer, I think we can copy the `c.opt` to `c.optReplica` with the `ReplicaOnly` set to true and then we can use `c.connFn(addr, c.optReplica)` for replica connections.

Reply:
[98040d7](https://github.com/redis/rueidis/pull/430/commits/98040d775392220820aea58a2738889c1607ad53)

@rueian 
I'm sorry about serious misunderstanding. I figured out.